From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Casey Schaufler <casey@schaufler-ca.com>
Date: Wed, 28 Jun 2023 15:13:30 -0700
Subject: [PATCH 27/90] UBUNTU: SAUCE: apparmor4.0.0 [27/90]: LSM stacking v39:
 LSM: Remove unused lsmcontext_init()

BugLink: http://bugs.launchpad.net/bugs/2028253

The lsmcontext init functing is no longer used.
Remove it.

Signed-off-by: Casey Schaufler <casey@schaufler-ca.com>
(cherry picked from commit 2eb231bf638779144b0b91678999a00474772cbf
https://git.launchpad.net/~apparmor-dev/ubuntu-kernel-next)
Signed-off-by: Paolo Pisati <paolo.pisati@canonical.com>
---
 include/linux/security.h | 19 -------------------
 1 file changed, 19 deletions(-)

diff --git a/include/linux/security.h b/include/linux/security.h
index be82ad366bad..721bfb421179 100644
--- a/include/linux/security.h
+++ b/include/linux/security.h
@@ -160,25 +160,6 @@ struct lsmcontext {
 	int	id;		/* Identifies the module */
 };
 
-/**
- * lsmcontext_init - initialize an lsmcontext structure.
- * @cp: Pointer to the context to initialize
- * @context: Initial context, or NULL
- * @size: Size of context, or 0
- * @id: Which LSM provided the context
- *
- * Fill in the lsmcontext from the provided information.
- * This is a scaffolding function that will be removed when
- * lsmcontext integration is complete.
- */
-static inline void lsmcontext_init(struct lsmcontext *cp, char *context,
-				   u32 size, int id)
-{
-	cp->id = id;
-	cp->context = context;
-	cp->len = size;
-}
-
 /*
  * Data exported by the security modules
  */
